core/src/Kokkos_Atomic.hpp

```diff
@@ -69,6 +69,15 @@
 #define KOKKOS_ATOMIC_HPP
 
 #include <Kokkos_Macros.hpp>
+
+#ifdef KOKKOS_ENABLE_IMPL_DESUL_ATOMICS
+#ifdef KOKKOS_ENABLE_OPENMPTARGET
+#define DESUL_HAVE_OPENMP_ATOMICS
```

Comment:
yes and no. In principle I agree, but for OpenMP its complicated. By default I actually want to use the GCC atomics even if OpenMP is enabled since they are more flexible (i.e. they got memory order, which OpenMP doesn't). But OpenMPTarget doesn't eat the GCC intrinsics properly in all cases. And I can't figure out from macros whether or not you compile for OpenMP offload or not. Kokkos knows it however.